<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1326</width>
    <height>629</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLineEdit" name="lineEditNPrivate">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>240</y>
      <width>151</width>
      <height>26</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEditN">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>170</y>
      <width>151</width>
      <height>26</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>270</y>
      <width>63</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Bản mã</string>
    </property>
   </widget>
   <widget class="QPushButton" name="selectFileSendButton">
    <property name="geometry">
     <rect>
      <x>690</x>
      <y>60</y>
      <width>93</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>Chọn file</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>60</y>
      <width>91</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Số nguyên tố </string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEditP">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>60</y>
      <width>151</width>
      <height>26</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_11">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>150</y>
      <width>131</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Khoá công khai (e,n)</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>60</y>
      <width>63</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>p=</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>90</y>
      <width>91</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Số nguyên tố</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_14">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>220</y>
      <width>131</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Khoá bí mật (d,n)</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>60</y>
      <width>63</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Văn bản</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_13">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>170</y>
      <width>63</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>n=</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_16">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>240</y>
      <width>63</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>n=</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>60</y>
      <width>63</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Văn bản</string>
    </property>
   </widget>
   <widget class="QPushButton" name="verifySignatureButton">
    <property name="geometry">
     <rect>
      <x>1020</x>
      <y>348</y>
      <width>111</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>Xác nhận</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_15">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>210</y>
      <width>63</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>d=</string>
    </property>
   </widget>
   <widget class="QPushButton" name="resetButton">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>450</y>
      <width>93</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>Reset</string>
    </property>
   </widget>
   <widget class="QPushButton" name="generateKeysButton">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>308</y>
      <width>111</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>Tạo khoá</string>
    </property>
   </widget>
   <widget class="QPushButton" name="exitButton">
    <property name="geometry">
     <rect>
      <x>690</x>
      <y>450</y>
      <width>93</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>Thoát</string>
    </property>
   </widget>
   <widget class="QPushButton" name="createSignatureButton">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>160</y>
      <width>93</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>Tạo chữ ký</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEditQ">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>90</y>
      <width>151</width>
      <height>26</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEditEncrypted">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>240</y>
      <width>141</width>
      <height>91</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="selectFileReceiveButton">
    <property name="geometry">
     <rect>
      <x>1130</x>
      <y>60</y>
      <width>93</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>Chọn file</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>1030</x>
      <y>30</y>
      <width>63</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Giải mã</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEditDecrypted">
    <property name="geometry">
     <rect>
      <x>980</x>
      <y>230</y>
      <width>141</width>
      <height>91</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>230</y>
      <width>63</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Bản rõ</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_12">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>140</y>
      <width>63</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>e=</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEditE">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>140</y>
      <width>151</width>
      <height>26</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="saveButton">
    <property name="geometry">
     <rect>
      <x>690</x>
      <y>120</y>
      <width>91</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>Lưu</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEditD">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>210</y>
      <width>151</width>
      <height>26</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>590</x>
      <y>30</y>
      <width>63</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Mã hóa</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>90</y>
      <width>63</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>q=</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEditSend">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>60</y>
      <width>141</width>
      <height>91</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEditReceive">
    <property name="geometry">
     <rect>
      <x>980</x>
      <y>60</y>
      <width>141</width>
      <height>91</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEditSignature">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>200</y>
      <width>141</width>
      <height>26</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="encryptButton">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>348</y>
      <width>111</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>Mã hoá</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_17">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>170</y>
      <width>63</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Chữ ký</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEditSignatureReceived">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>980</x>
      <y>170</y>
      <width>141</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="Line" name="line">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>20</y>
      <width>20</width>
      <height>361</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Orientation::Vertical</enum>
    </property>
   </widget>
   <widget class="Line" name="line_5">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>370</y>
      <width>341</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Orientation::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_6">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>20</y>
      <width>20</width>
      <height>361</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Orientation::Vertical</enum>
    </property>
   </widget>
   <widget class="Line" name="line_7">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>341</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Orientation::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_2">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>20</y>
      <width>20</width>
      <height>401</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Orientation::Vertical</enum>
    </property>
   </widget>
   <widget class="Line" name="line_3">
    <property name="geometry">
     <rect>
      <x>780</x>
      <y>20</y>
      <width>20</width>
      <height>401</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Orientation::Vertical</enum>
    </property>
   </widget>
   <widget class="Line" name="line_4">
    <property name="geometry">
     <rect>
      <x>880</x>
      <y>20</y>
      <width>20</width>
      <height>401</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Orientation::Vertical</enum>
    </property>
   </widget>
   <widget class="Line" name="line_8">
    <property name="geometry">
     <rect>
      <x>1240</x>
      <y>20</y>
      <width>20</width>
      <height>401</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Orientation::Vertical</enum>
    </property>
   </widget>
   <widget class="Line" name="line_9">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>410</y>
      <width>351</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Orientation::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_10">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>10</y>
      <width>351</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Orientation::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_11">
    <property name="geometry">
     <rect>
      <x>890</x>
      <y>10</y>
      <width>361</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Orientation::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_12">
    <property name="geometry">
     <rect>
      <x>890</x>
      <y>410</y>
      <width>361</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Orientation::Horizontal</enum>
    </property>
   </widget>
   <widget class="QPushButton" name="exportSignatureButton">
    <property name="geometry">
     <rect>
      <x>690</x>
      <y>200</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Xuất chữ ký</string>
    </property>
   </widget>
   <widget class="QPushButton" name="importSignatureButton">
    <property name="geometry">
     <rect>
      <x>1130</x>
      <y>170</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Import chữ ký</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_18">
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>200</y>
      <width>63</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Chữ ký</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1326</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
